cmd/ctr/app/main.go

```diff
@@ -119,11 +129,44 @@ containerd CLI
 		sandboxes.Command,
 		info.Command,
 	}, extraCmds...)
-	app.Before = func(context *cli.Context) error {
-		if context.GlobalBool("debug") {
+
+	var traceProvider *trace.TracerProvider
+
+	app.Before = func(cctx *cli.Context) error {
+		if cctx.GlobalBool("debug") {
 			logrus.SetLevel(logrus.DebugLevel)
 		}
+
+		if !cctx.GlobalBool("opentelemetry") {
+			return nil
+		}
+
+		ctx := context.Background()
+
+		exp, err := otlptracehttp.New(ctx)
+		if err != nil {
+			return err
+		}
+		res, err := resource.New(ctx,
+			resource.WithHost(),
+			resource.WithAttributes(semconv.ServiceNameKey.String("ctr")),
+		)
+		if err != nil {
+			return err
+		}
+		traceProvider := trace.NewTracerProvider(
+			trace.WithSpanProcessor(trace.NewSimpleSpanProcessor(exp)),
+			trace.WithResource(res),
+		)
+		otel.SetTracerProvider(traceProvider)
```

Comment:
what happens if containerd tracing plugin is already initialized?	`otel.SetTracerProvider(traceProvider)` sets the global trace provider, so will this overwrite the provider set during plugin init?

Reply:
ctr and containerd are separate binaries and ctr doesn't include containerd plugins.